yb-voyager/cmd/analyzeSchema.go

```diff
@@ -593,6 +593,72 @@ func checker(sqlInfoArr []sqlInfo, fpath string, objType string) {
 	}
 }
 
+func checkRedundantIndexes() {
+	msr, err := metaDB.GetMigrationStatusRecord()
+	if err != nil {
+		utils.ErrExit("error getting migration status record: %v", err)
+	}
+	if !msr.MigrationAssessmentDone { //TODO: add the case to handle if assessment is run in export schema
+		return
+	}
+
+	if msr.SourceDBConf.DBType != POSTGRESQL {
+		return
+	}
+
+	//Only considering the case where assessment is run in this migration
+	//If not via assess-migration, there is going to be a change where internally in the export-schema we will run assessment
+	//so we should have this information most of the time.
+	assessmentReportPath := filepath.Join(exportDir, "assessment", "reports", fmt.Sprintf("%s.json", ASSESSMENT_FILE_NAME))
+	if !utils.FileOrFolderExists(assessmentReportPath) {
+		log.Infof("migration assessment report file doesn't exists at %q, skipping apply recommendations step...", assessmentReportPath)
+		return
+	}
+
+	log.Infof("parsing assessment report json file for redundant indexes")
+	report, err := ParseJSONToAssessmentReport(assessmentReportPath)
+	if err != nil {
+		utils.ErrExit("error parsing json report file %q: %w", assessmentReportPath, err)
+	}
+	assessmentIssues := report.Issues
+	redundantIndexIssues := lo.Filter(assessmentIssues, func(i AssessmentIssue, _ int) bool {
+		return i.Category == PERFORMANCE_OPTIMIZATIONS_CATEGORY && i.Name == queryissue.REDUNDANT_INDEXES_ISSUE_NAME
+	})
+	for _, redundantIndexIssue := range redundantIndexIssues {
+		//this is the Unquoted object name
+		indexObjectName := redundantIndexIssue.ObjectName
+		//this is to create a proper objectname with Quotes if required to do the check in the objSet of Index if its present or not.
+		splits := strings.Split(indexObjectName, " ON ")
+		indexName := splits[0]
+		qualifiedTableName := ""
+		if len(splits) > 1 {
+			qualifiedTableName = splits[1]
+		}
+		tableObjectName := sqlname.NewObjectNameWithQualifiedName(POSTGRESQL, "", qualifiedTableName)
+		indexSQlObjectName := sqlname.NewObjectName(POSTGRESQL, "", tableObjectName.SchemaName, indexName)
+		qualifiedIndexName := fmt.Sprintf(`%s ON %s`, indexSQlObjectName.Unqualified.MinQuoted, tableObjectName.MinQualified.MinQuoted)
```

Comment:
Can we save it this way (i.e. qualified) while generating the assessment report itself? Rather than doing all this at this point? 

Let's discuss this once on call.